src/components/Post.jsx

```diff
@@ -288,6 +293,36 @@ function Post(prop) {
             >
               <MoreHorizOutlinedIcon />
             </IconButton>
+            <Menu
+              id="long-menu"
+              MenuListProps={{
+                "aria-labelledby": "long-button",
+              }}
+              anchorEl={anchorEl}
+              open={open}
+              onClose={() => setAnchorEl(null)}
+              PaperProps={{
+                style: {
+                  maxHeight: ITEM_HEIGHT * 4.5,
+                  width: "20ch",
+                },
+              }}
+            >
+              <MenuItem onClick={handleDownload}> Download </MenuItem>
+              <MenuItem
+                onClick={() => {
+                  navigate("/dummygram/profile", {
+                    state: {
+                      name: username,
+                      avatar: avatar,
+                    },
+                  });
+                }}
+              >
+                {" "}
+                Visit Profile{" "}
+              </MenuItem>
+            </Menu>
             {user && username == user.displayName && (
```

Comment:
shift this conditional checking of line 326 to menuItem level (line 342)so you don't have to duplicate menu element